src/test/java/persistence/study/ReflectionTest.java

```diff
@@ -0,0 +1,144 @@
+        assertThat(Arrays.stream(declaredFields).count()).isEqualTo(2);
+    }
+
+    @Test
+    @DisplayName("Car 생성자들을 가져오기")
+    void showConstructors() {
+        Class<Car> carClass = Car.class;
+        Constructor<?>[] constructors = carClass.getConstructors();
+        assertThat(Arrays.stream(constructors).count()).isEqualTo(2);
+    }
+
+    @Test
+    @DisplayName("Car 메서드들을 가져오기")
+    void showMethods() {
+        Class<Car> carClass = Car.class;
+        List<String> actual = Arrays.stream(carClass.getMethods())
+                .map(Method::getName)
+                .collect(Collectors.toList());
+        assertThat(actual).contains("testGetName", "printView");
+    }
+
+    @Test
+    @DisplayName("test로 시작하는 메서드를 실행한다")
+    void executeMethodsStartingWithTest() throws Exception {
+        Class<Car> carClass = Car.class;
+        Method[] declaredMethods = carClass.getDeclaredMethods();
+        List<Method> test = Arrays.stream(declaredMethods)
+                .filter(it -> it.getName().startsWith("test"))
+                .collect(Collectors.toList());
+        Car car = carClass.getDeclaredConstructor().newInstance();
+
+        for (Method method : test) {
+            LOGGER.debug(String.valueOf(method));
+            Object invoke = method.invoke(car);
+        }
+    }
+
+    @Test
+    @DisplayName("@PrintView 애노테이션 메서드 실행한다.")
+    void executeMethodsWithPritViewAnnotation() throws Exception {
+        Class<Car> carClass = Car.class;
+
+        List<Method> printViewAnnotatedMethods = Arrays.stream(carClass.getDeclaredMethods())
+                .filter(it -> it.isAnnotationPresent(PrintView.class))
+                .collect(Collectors.toList());
+        Car car = carClass.getDeclaredConstructor().newInstance();
+
+        for (Method printViewAnnotatedMethod : printViewAnnotatedMethods) {
+            LOGGER.debug(String.valueOf(printViewAnnotatedMethod));
+            printViewAnnotatedMethod.invoke(car);
+        }
+    }
+
+    @Test
+    @DisplayName("private field에 값을 할당할 수 있다.")
+    void insertValueInPrivateField() throws Exception {
+        Class<Car> carClass = Car.class;
+        Car car = carClass.getDeclaredConstructor().newInstance();
+
+        Field[] declaredFields = carClass.getDeclaredFields();
+        List<Field> privateFields = Arrays.stream(declaredFields)
+                .filter(it -> Modifier.isPrivate(it.getModifiers()))
+                .collect(Collectors.toList());
+        for (Field privateField : privateFields) {
+            privateField.setAccessible(true);
+            Class<?> type = privateField.getType();
+            privateField.set(car, generateTestValue(type));
+        }
+        int expectedPrice = 1000;
+        String expectedName = "carName";
+        assertAll(
+                () -> assertThat(car.getPrice()).isEqualTo(expectedPrice),
+                () -> assertThat(car.getName()).isEqualTo(expectedName)
+        );
```

Comment:
```suggestion
        assertAll(
                () -> assertThat(car.getPrice()).isEqualTo(1000),
                () -> assertThat(car.getName()).isEqualTo("carName")
        );
```

저는 객체지향에서 `객체에게 물어보라`라는 말을 좋아하는데요 ㅎ
[TDA 원칙](https://effectiveprogramming.tistory.com/entry/Tell-dont-ask)이라고도 해요.

값을 꺼내와서 비교하기 보다는 비교하라고 시켜보면 어떨까요?
`assertThat(car.가격같아?(expectedPrice)).isTrue(),`
테스트 코드도 더 간결해지는것 같에요 ㅎ

물론, 1단계에서는 위 처럼 구현하는게 더 자연스럽습니다 ^^
앞으로 과정 진행하면서 위 TDA 내용을 참고 하셔서 진행하시면 좀 더 객체지향적 사고에 다가살수 있을것 같아서 코멘트 남겨봅니다! 😄

Reply:
의견 감사합니다ㅎㅎ
저도 태화님 의견에 동의합니다.
Reflection 학습테스트가 실제 프로덕션에 코드를 만들어두고 하지 않다보니 이런식의 코드가 조금 작성된 것 같습니다ㅎㅎ
미션을 진행하면서는 커멘트 해주신 내용처럼 진행하도록 하겠습니다!